### PR TITLE
Add compat with hls.js v1

### PR DESCRIPTION
**Pull request by Chocobozzz**
Fixes https://github.com/Novage/p2p-media-loader/issues/176

### CODE REVIEW

**Comment by askz** [comment]
up !

Is the project dead?

**Comment by hoangvietfreelancer** [comment]
Thanks for adding support for hls.js v1!
It's all working if I add non-AES encrypt content to JWPlayer.
But when add AES encrypt to JWPlayer, I got this problem:
```
Uncaught TypeError: n.onProgress is not a function
    at r.HlsJsLoader.successSegment (p2p-media-loader-hlsjs.min.js:formatted:208)
    at p2p-media-loader-hlsjs.min.js:formatted:165
```

File: p2p-media-loader-hlsjs/lib/hlsjs-loader.ts [hunk 1]
```diff
@@ -80,7 +101,15 @@ export class HlsJsLoader {
...
         this.stats.tload = now;
+        this.stats.loading = {
+            first: this.stats.tfirst,
+            start: this.stats.trequest,
+            end: this.stats.tload
+        };
         this.stats.loaded = content.byteLength;
+        this.stats.total = content.byteLength;
+
+        callbacks.onProgress(this.stats, context, content);
```
Comment (hoangvietfreelancer) [inline p2p-media-loader-hlsjs/lib/hlsjs-loader.ts:112]:
```suggestion
        if(callbacks.onProgress) {
            callbacks.onProgress(this.stats, context, content);
        }
```
AES key file does not require onProgress callback so adding this check will prevent ```onProgress is not a function``` error

**Comment by Teranode** [comment]
Have you tested this with debug enabled?

**Review by hoangvietfreelancer — CHANGES_REQUESTED** [review]
hls.js v1 add ```rangeStart: 0``` and ```rangeEnd: 0``` to xhr as default instead of nothing(undefined) so we need to change this line:
https://github.com/Novage/p2p-media-loader/blob/e661cabde09f2790ac12d5b499f980fba2467b42/p2p-media-loader-hlsjs/lib/hlsjs-loader.ts#L72
to
```typescript
((context.rangeStart == undefined) || (context.rangeEnd == undefined)) || (context.rangeStart === 0 && context.rangeEnd === 0)
```
to make it work with p2p(otherwise, it only use HTTP)

**Comment by Chocobozzz** [comment]
Thanks @hoangvietfreelancer, I updated the PR. Please test and tell me if it's working on your side.

**Comment by mrlika** [comment]
> up !
> 
> Is the project dead?

No! It is still moving! But very slow during last year...

@Chocobozzz, thanks for the PR. I am reviewing hls.js v1 support now. I slightly improved types usage for hls.js < v1 which makes further development and support more reliable. I will try to do it in a similar way for v1 taking this PR as a reference implementation.

**Comment by piranna** [comment]
If I can help to move this fordward, just tell me :-)

**Comment by mrlika** [comment]
@piranna , you can help by testing this branch:

https://github.com/Novage/p2p-media-loader/tree/feat/hlsjs1

it implements almost the same approach as in this PR but applied to the latest code that uses hls.js type definitions.

**Comment by piranna** [comment]
> @piranna , you can help by testing this branch:

Great, I'll take a look :-)

**Comment by Chocobozzz** [comment]
Closing now it's being worked on a dedicated branch. Thanks @mrlika

**Comment by mrlika** [comment]
@Chocobozzz ,  it mostly works but for some reason bitrate/quality is chosen lower than it should be even when hard-coded to a max value. Requires time to investigate why.

**Comment by Chocobozzz** [comment]
May be related to recent prettier/eslint changes on master. You renamed `byterange` to `byteRange` but `m3u8-parser` returns segments with `byterange` property.

**Comment by kontrollanten** [comment]
@mrlika Is everything working now that #202 is merged? Is there anything I can do to help releasing hls.js v1 support?

**Comment by mrlika** [comment]
@kontrollanten, it works with issues. You can join the development: https://github.com/Novage/p2p-media-loader/pull/203

**Pull request closed without merging**